conda_build/render.py

```diff
@@ -200,37 +201,36 @@ def _filter_run_exports(specs, ignore_list):
     return filtered_specs
 
 
-def find_pkg_dir_or_file_in_pkgs_dirs(pkg_dist, m, files_only=False):
-    _pkgs_dirs = pkgs_dirs + list(m.config.bldpkgs_dirs)
-    pkg_loc = None
-    for pkgs_dir in _pkgs_dirs:
-        pkg_dir = os.path.join(pkgs_dir, pkg_dist)
-        pkg_file = os.path.join(pkgs_dir, pkg_dist + CONDA_PACKAGE_EXTENSION_V1)
-        if not files_only and os.path.isdir(pkg_dir):
-            pkg_loc = pkg_dir
-            break
-        elif os.path.isfile(pkg_file):
-            pkg_loc = pkg_file
-            break
-        elif files_only and os.path.isdir(pkg_dir):
-            pkg_loc = pkg_file
-            # create the tarball on demand.  This is so that testing on archives works.
-            with tarfile.open(pkg_file, 'w:bz2') as archive:
-                for entry in os.listdir(pkg_dir):
-                    archive.add(os.path.join(pkg_dir, entry), arcname=entry)
-
-            # use the package's subdir
+def find_pkg_dir_or_file_in_pkgs_dirs(
+    distribution: str, m: MetaData, files_only: bool = False
+) -> str | None:
```

Comment:
I don't think this can be used in the return type, but only in argument declarations. 

```suggestion
) -> Optional[str]:
```

Reply:
per the docs the union type is allowed anywhere we used to say `Optional`:
- https://docs.python.org/3/library/typing.html#typing.Optional
- https://docs.python.org/3/library/stdtypes.html#types-union